Wordle.Api/Wordle.Api/Services/LeaderboardService.cs

```diff
@@ -0,0 +1,76 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Wordle.Api.Data;
+using Wordle.Api.Dtos;
+using Wordle.Api.Models;
+
+namespace Wordle.Api.Services;
+
+public class LeaderboardService
+{
+	private readonly AppDbContext _context;
+	private static object _changingPlayerLock = new();
+	private static object _addingPlayerLock = new();
+	public LeaderboardService(AppDbContext context)
+	{
+		_context = context;
+	}
+	public async Task<List<PlayerDto>> GetTopScoresAsync()
+	{
+		return await _context.Players.
+			OrderBy(player => player.AverageAttempts).
+			Take(10).
+			Select(player => new PlayerDto
+			{
+				Name = player.Name,
+				GameCount = player.GameCount,
+				AverageAttempts = player.AverageAttempts
+			}).ToListAsync();
+	}
+
+	public async Task<Player> PostScoreAsync(PlayerDto request)
+	{
+		Player? foundPlayer = await _context.Players.FirstOrDefaultAsync(player => player.Name.Equals(request.Name));
+		if (foundPlayer is not null)
+		{
+			lock (_changingPlayerLock)
+			{
+				double attempts = foundPlayer.AverageAttempts * foundPlayer.GameCount + request.AverageAttempts;
+				foundPlayer.GameCount += request.GameCount;
+				foundPlayer.AverageAttempts = attempts / foundPlayer.GameCount;

```

Comment:
You may be able to calculate the updated average seconds per game for a player in your leaderboard system here with something like this: 
```  
foundPlayer.AverageSecondsPerGame = ((foundPlayer.AverageSecondsPerGame * (foundPlayer.GameCount - 1)) + request.AverageSecondsPerGame) / foundPlayer.GameCount;
```

Make sure that the client-side logic (maybe in your game's end logic) captures the time taken per game and sends this data along with the score data.

Reply:
Definitely! Thank you